### PR TITLE
Transitions are still valid with only the route name, models are optional

### PR DESCRIPTION
**Pull request by mfeckie**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/emberjs/ember.js/blob/v2.15.3/packages/ember-routing/lib/system/router.js#L364

There's some weirdness with the linter that is unrelated to anything I've changed in this PR

### CODE REVIEW

File: types/ember/index.d.ts [hunk 1]
```diff
@@ -1950,6 +1950,7 @@ declare module 'ember' {
              * Transition the application into another route. The route may
              * be either a single route or route path
              */
+            transitionTo(name: string): Transition;
```
Comment (dwickern) [inline types/ember/index.d.ts]: Are you sure this is needed? You should be able to call the vararg overload with 1 argument.

File: types/ember/index.d.ts [hunk 2]
```diff
@@ -3339,6 +3342,7 @@ declare module 'ember' {
          *                       containing a mapping of query parameters
          * @returns              the Transition object associated with this attempted transition
          */
+        transitionTo(routeNameOrUrl: string): Ember.Transition;
```
Comment (dwickern) [inline types/ember/index.d.ts]: This method should accept an optional `options` as well.
Could you also add no-model overloads to the other router service methods?

**Review by dwickern — CHANGES_REQUESTED** [review]

**Comment by andy-ms** [comment]
@mfeckie Please address comments from the code reviewers.

File: types/ember/index.d.ts (hunk 1 above)
Comment (mfeckie) [reply types/ember/index.d.ts]: Yep, you're right, removing.

File: types/ember/index.d.ts (hunk 2 above)
Comment (mfeckie) [reply types/ember/index.d.ts]: Will do

**Comment by mfeckie** [comment]
@dwickern I've been on holidays, back now.  Have reworked this PR and added to some tests to consume the `RouterService`

**Review by dwickern — APPROVED** [review]

**Pull request merged**